server/task/src/test/java/org/apache/james/task/MemoryTaskManagerTest.java

```diff
@@ -92,6 +92,7 @@ void taskCodeAfterCancelIsNotRun() {
         TaskId id = memoryTaskManager.submit(() -> {
             waitForTaskToBeLaunched.countDown();
             await(task1Latch);
+            Thread.sleep(1);
```

Comment:
I guess it deserves a comment :)

Reply:
it suppose to not increase counter but actually the counter is increased after calling cancel(). Why didn't you fix the cancel method instead?

Reply:
> Why didn't you fix the cancel method instead?

Because, in my knowledge, it's impossible. Interrupting a task uses thread JVM interruption mechanism. And this mechanism doesn't guarantee the thread stop immediately, it just promises to try to stop it as soon as possible.

The sleep call may (I'm not sure about that) give the JVM the opportunity to interrupt the thread because it handover its running status to the JVM.

I'm not sure it's work. The alternative is to remove this test if we can't find a way to enforce such a high expectation.

Reply:
1 ms seems short?